File: BaekJoon Online Judge/C++/Step12-Dynamic Programming_1/2748.cpp
//
//  2748.cpp
//  Step12-Dynamic Programming_1
//
//  Created by Yebin Kim on 2019/10/25.
//  Copyright © 2019 김예빈. All rights reserved.
//

#include <iostream>

using namespace std;

// Time Complexity O(n)
int main() {
    cin.tie(NULL);
    ios::sync_with_stdio(false);
    
    int n;
    
    cin >> n;
    
    long dp[n + 1];
    
    dp[0] = 0;
    dp[1] = 1;
    
    for(int i = 2; i <= n; i++)
        dp[i] = dp[i - 1] + dp[i - 2];
    
    cout << dp[n];
    
    return 0;
}

File: BaekJoon Online Judge/Swift/etc/2748.swift
//
//  2748.swift
//  Step14-Dynamic Programming_1
//
//  Created by Yebin Kim on 2019/12/06.
//  Copyright © 2019 김예빈. All rights reserved.
//

import Foundation

if let N = Int(readLine()!) {
    var dp = Array<Int>(stride(from: 1, to: N + 1, by: 1))
    
    for i in dp {
        dp[i - 1] = i < 3 ? 1 : dp[i - 2] + dp[i - 3]
    }
    
    print(dp.last!)
}
